arch/x86/src/boundary/boundary_impl.rs

```diff
@@ -117,59 +112,26 @@ impl UserspaceKernelBoundary for Boundary {
             &kernel::utilities::arch_helpers::TRD104SyscallReturn::from_syscall_return(
                 return_value,
             ),
-            &mut ret0,
-            &mut ret1,
-            &mut ret2,
-            &mut ret3,
+            &mut state.ebx,
+            &mut state.ecx,
+            &mut state.edx,
+            &mut state.edi,
```

Comment:
The big block comment above this about pointer safety seems like it's no-longer relevant since these aren't on the stack.